src/Microsoft.SqlTools.ServiceLayer/FileBrowser/FileBrowserService.cs

```diff
@@ -158,14 +159,26 @@ public void InitializeService(ServiceHost serviceHost)
             RequestContext<FileBrowserCloseResponse> requestContext)
         {
             FileBrowserCloseResponse response = new FileBrowserCloseResponse();
-            if (this.ownerToFileBrowserMap.ContainsKey(fileBrowserParams.OwnerUri))
+            try
             {
-                this.ownerToFileBrowserMap.Remove(fileBrowserParams.OwnerUri);
-                response.Succeeded = true;
+                if (this.ownerToFileBrowserMap.ContainsKey(fileBrowserParams.OwnerUri))
+                {
+                    FileBrowserOperation browserOperation = this.ownerToFileBrowserMap[fileBrowserParams.OwnerUri];
+                    if (!browserOperation.FileTreeCreated)
+                    {
+                        browserOperation.Cancel();
```

Comment:
so cancel doesn't close the connection?

Reply:
Before it was handling connection close in open task if the file tree was not created yet, but I simplified it and close the connection in Close request task all the time..